core/src/main/java/org/apache/brooklyn/core/mgmt/persist/XmlMementoSerializer.java

```diff
@@ -454,25 +460,25 @@ public Object unmarshal(HierarchicalStreamReader reader, UnmarshallingContext co
                 result.catalogItemId(catalogItemId);
                 return result;
             } finally {
-                instance = null;
+                SpecConverterLocalInstance.remove();
                 if (customLoaderSet) {
                     popXstreamCustomClassLoader();
                 }
             }
         }
 
-        Object instance;
-        
         @Override
         protected Object instantiateNewInstance(HierarchicalStreamReader reader, UnmarshallingContext context) {
             // the super calls getAttribute which requires that we have not yet done moveDown,
             // so we do this earlier and cache it for when we call super.unmarshal
+            Object instance = SpecConverterLocalInstance.get();
```

Comment:
I believe you can put arbitrary objects in the `context`, just for cases like this.